.github/workflows/build.yml

```diff
@@ -2,29 +2,51 @@ name: Build
 
 on:
   push:
-    branches: [ main ]
+    branches: [main]
   pull_request:
     branches: '*'
 
+env:
+  CACHE_EPOCH: 0
+
 jobs:
   build:
     runs-on: ubuntu-latest
     steps:
-    - name: Checkout
-      uses: actions/checkout@v2
-    - name: Install node
-      uses: actions/setup-node@v2
-      with:
-       node-version: '14.x'
-    - name: Build
-      run: |
-        yarn
-        yarn run build
-    - name: Lint
-      run: |
-        yarn run eslint:check
-        yarn run prettier:check
-    - name: Test
-      run: |
-        yarn run build:test
-        yarn run test
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Install node
+        uses: actions/setup-node@v2
+        with:
+          node-version: '14.x'
+      - name: Cache node modules
+        uses: actions/cache@v2
+        id: cache-node-modules
+        with:
+          path: 'node_modules/'
+          key: |
+            ${{ env.CACHE_EPOCH }}-${{ runner.os }}-node-modules-${{ hashFiles('yarn.lock') }}
+      - name: Cache yarn packages
+        uses: actions/cache@v2
+        id: cache-yarn-packages
+        if: steps.cache-node-modules.outputs.cache-hit != 'true'
+        with:
+          path: '.yarn-packages'
+          key: |
+            ${{ env.CACHE_EPOCH }}-yarn-packages-${{ hashFiles('yarn.lock') }}
+          restore-keys: |
+            ${{ env.CACHE_EPOCH }}-yarn-packages-
+      - name: Install
+        if: steps.cache-node-modules.outputs.cache-hit != 'true'
```

Comment:
had forgotten this one to get the actual benefit of caching... knocks a minute off the runtime.